geopandas/tools/sjoin.py

```diff
@@ -329,3 +333,161 @@ def _frame_join(indices, left_df, right_df, how, lsuffix, rsuffix):
+    how : string, default 'inner'
+        The type of join:
+
+        * 'left': use keys from left_df; retain only left_df geometry column
+        * 'right': use keys from right_df; retain only right_df geometry column
+        * 'inner': use intersection of keys from both dfs; retain only
+          left_df geometry column
+    max_distance : float, default None
+        Maximum distance within which to query for nearest geometry.
+        Must be greater than 0.
+        The max_distance used to search for nearest items in the tree may have a
+        significant impact on performance by reducing the number of input
+        geometries that are evaluated for nearest items in the tree.
+    lsuffix : string, default 'left'
+        Suffix to apply to overlapping column names (left GeoDataFrame).
+    rsuffix : string, default 'right'
+        Suffix to apply to overlapping column names (right GeoDataFrame).
+    distance_col : string, default None
+        If set, save the distances computed between matching geometries under a
+        column of this name in the joined GeoDataFrame.
+
+    Examples
+    --------
+    >>> countries = geopandas.read_file(geopandas.datasets.get_\
+path("naturalearth_lowres"))
+    >>> cities = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))
+    >>> countries.head(2)  # doctest: +SKIP
+        pop_est      continent                      name \
+iso_a3  gdp_md_est                                           geometry
+    0     920938        Oceania                      Fiji    FJI      8374.0  MULTIPOLY\
+GON (((180.00000 -16.06713, 180.00000...
+    1   53950935         Africa                  Tanzania    TZA    150600.0  POLYGON (\
+(33.90371 -0.95000, 34.07262 -1.05982...
+    >>> cities.head(2)  # doctest: +SKIP
+            name                   geometry
+    0  Vatican City  POINT (12.45339 41.90328)
+    1    San Marino  POINT (12.44177 43.93610)
+
+    >>> cities_w_country_data = geopandas.sjoin_nearest(cities, countries)
+    >>> cities_w_country_data.head(2)  # doctest: +SKIP
+            name_left                   geometry  index_right   pop_est continent name_\
+right iso_a3  gdp_md_est
+    0    Vatican City  POINT (12.45339 41.90328)          141  62137802    Europe      \
+Italy    ITA   2221000.0
+    1      San Marino  POINT (12.44177 43.93610)          141  62137802    Europe      \
+Italy    ITA   2221000.0
+    To include the distances:
+    >>> countries = geopandas.read_file(geopandas.datasets.get_\
+path("naturalearth_lowres"))
+    >>> cities = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))
+    >>> cities_w_country_data = geopandas.sjoin_nearest\
+(cities, countries, distance_col="distances")
+    >>> cities_w_country_data[["name_left", "name_right", \
+"distances"]].head(2)  # doctest: +SKIP
+            name_left name_right distances
+    0    Vatican City      Italy       0.0
+    1      San Marino      Italy       0.0
+
```

Comment:
It might be a good idea to show the opposite direction, either using `how='right'` (after #2064 is resolved) or by flipping the left and right data frames.  Specifically, the goal is to show that there are 3 results for Italy due to 3 intersecting cities with a distance of 0.

Reply:
I don't think we can easily do that easily just flipping the dfs and using parameters. I added an example where I manually filter out the results for Italy and explain them.